.golangci.yml

```diff
@@ -1,23 +1,70 @@
-run:
-  concurrency: 4
-
-  modules-download-mode: readonly
-
 linters-settings:
+  exhaustive:
+    default-signifies-exhaustive: false
+  goimports:
+    local-prefixes: github.com/kubernetes-sigs/external-dns
   golint:
-      min-confidence: 0.9
-
-  gocyclo:
-      min-complexity: 15
+    min-confidence: 0.9
+  maligned:
+    suggest-new: true
+  misspell:
+    locale: US
 
 linters:
+  # please, do not use `enable-all`: it's deprecated and will be removed soon.
+  # inverted configuration with `enable-all` and `disable` is not scalable during updates of golangci-lint
   disable-all: true
   enable:
+    - deadcode
+    - depguard
+    - dogsled
+    - gofmt
+    - goimports
+    - golint
+    - goprintffuncname
+    - gosimple
     - govet
     - ineffassign
-    - golint
-    - goimports
+    - interfacer
     - misspell
+    - rowserrcheck
+    - staticcheck
+    - structcheck
+    - stylecheck
+    - typecheck
     - unconvert
-    - megacheck
-    - interfacer
+    - unused
+    - varcheck
+    - whitespace
+
+issues:
+  # Excluding configuration per-path, per-linter, per-text and per-source
+  exclude-rules:
+    - path: _test\.go
```

Comment:
@njuettner do we really want to disable all checks for the test code? Usually I prefer to think about test code to be equally important to production one.

Reply:
I agree we should not, however I would rather start with this approach and then as soon as possible add the linter for test files. The only problem which I faced was it is simply impossible to fix it within 2 hours. There were so many complaints in the test files which was too much to handle.

I would like to merge it because I want to have a linter which also checks if code it not used.

Reply:
got it, cool. I've created https://github.com/kubernetes-sigs/external-dns/issues/1621 to handle it later